ksqldb-engine/src/main/java/io/confluent/ksql/engine/generic/GenericExpressionResolver.java

```diff
@@ -76,11 +83,12 @@ public Object resolve(final Expression expression) {
 
     @Override
     protected Object visitExpression(final Expression expression, final Void context) {
+      new EnsureNoColReferences(expression).process(expression, context);
       final ExpressionMetadata metadata =
           CodeGenRunner.compileExpression(
```

Comment:
Just curious, how much overhead is it to compile the java into bytecode, load the bytecode, and then run it.  It makes a lot of sense when you compile once and then run many times, but in this case, there's a 1:1 relationship.  In general, it seems like interpreting the expression would be lower overhead and possibly faster.  Do we do it this way because this is the main method that currently exists for "evaluating a sql expression"?

I'd be curious to run a benchmark doing just expression lookups (rather than literals).

Reply:
Talked offline about this, we'll look into running specialized benchmarks to see how much the overhead is. Given that this is just "new" functionality, we won't block the PR on this and we'll run those benchmarks going forward.